### PR TITLE
task 3

### PR DESCRIPTION
**Pull request by DimaRomaniuk**
Created 2 lambda
endpoints:
  GET - https://hhs4y24seg.execute-api.eu-west-1.amazonaws.com/dev/product/{product_id}
  GET - https://hhs4y24seg.execute-api.eu-west-1.amazonaws.com/dev/products
added to fe repository
https://github.com/DimaRomaniuk/nodejs-aws-fe/commit/dba7d1b0c9dac8b8eaa5c6e3695a0062cf875015
https://github.com/DimaRomaniuk/nodejs-aws-fe/commit/fdf43b07e5559bd62ad3961e537199140fc3795f
Completed:
+1 - Async/await is used in lambda functions
+1 - ES6 modules are used for product-service implementation
+1 - Webpack is configured for product-service
+1 - Lambda handlers (getProductsList, getProductsById) code is written not in 1 single module (file) and separated in codebase.
+1 - Main error scenarious are handled by API ("Product not found" error, try catch blocks are used in lambda handlers).
+1 - Lambda handlers are covered by basic UNIT tests (NO infrastructure logic is needed to be covered) (You may use JEST)
+1 - SWAGGER documentation is created for product-service https://app.swaggerhub.com/apis/home8677/product-service/1.0.0

all requirements completed
total:  11/11

### CODE REVIEW

**Comment by Jadgptw** [comment]
Could you please provide a link to the CloundFront?

Also when I try to go to https://hhs4y24seg.execute-api.eu-west-1.amazonaws.com/dev/product/{product_id} link I receive "page can’t be found" instead of error "Product not found". Please fix this behavior

**Comment by Mrdev1ce** [comment]
Names of your APIs is not strictly follows to rules of REST API. It's better to follow next rule of naming:
model/{modelID}

In your case it could be:
/product and /product/{productId}
or
/products and /products/{productId}

**Comment by VarvaraZadnepriak** [comment]
Added comment for commit in FE repo: https://github.com/DimaRomaniuk/nodejs-aws-fe/commit/dba7d1b0c9dac8b8eaa5c6e3695a0062cf875015

File: product-service/handlers/product.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    'Access-Control-Allow-Headers': 'Origin, X-Requested-With, Content-Type, Accept',
+    'Access-Control-Allow-Methods': 'POST, GET, OPTIONS, PATCH, PUT',
+  }
+  try {
+    const data = (await productList).find(({ id }) => id === event.pathParameters.product_id)
+    if (!data) {
+      return {
+        statusCode: 404,
+        headers,
+        body: '',
```
Comment (VarvaraZadnepriak) [inline product-service/handlers/product.ts:18]: I suppose there should be an error text for 404

**Comment by VarvaraZadnepriak** [comment]
In swagger file (https://app.swaggerhub.com/apis/home8677/product-service/1.0.0) it would be nice to remove all references of pets, provide correct descriptions, host and remove parameters section for get all products endpoint

File: product-service/handlers/product.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import 'source-map-support/register';
+import productList from './productList';
+
+export const getById: APIGatewayProxyHandler = async (event, _context) => {
+  const headers = {
```
Comment (VarvaraZadnepriak) [inline product-service/handlers/product.ts:6]: Can extract to some shared constant between lambdas

File: product-service/handlers/products.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+export const getByType: APIGatewayProxyHandler = async () => {
+  const headers = {
+    'Content-Type': 'application/json',
+    'Access-Control-Allow-Origin': '*',
+    'Access-Control-Allow-Headers': 'Origin, X-Requested-With, Content-Type, Accept',
+    'Access-Control-Allow-Methods': 'POST, GET, OPTIONS, PATCH, PUT',
+  }
+  try {
+    const data = await productList
+    return {
```
Comment (VarvaraZadnepriak) [inline product-service/handlers/products.ts:14]: Can also extract error handling logic to some helpers to make it common for all lambdas

**Pull request merged**